Assets/__Scripts/UI/CM_ColorInputBox.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using UnityEngine;
+using TMPro;
+using System.Collections;
+using UnityEngine.EventSystems;
+using UnityEngine.InputSystem;
+using System.Linq;
+using System.Collections.Generic;
+
+public class CM_ColorInputBox : MenuBase
+{
+    [SerializeField] private ColorPicker ColorInputField;
+    [SerializeField] private TextMeshProUGUI UIMessage;
+    [SerializeField] private CanvasGroup group;
+    private Action<object> resultAction;
```

Comment:
Why is this object and not Color?

Reply:
This was because I was following what the other dialog boxes do. They invoke `result` with `null` when cancelled, and Color is a non-nullable type, so invoking with null wouldn't work. If there is no reason for the input box to invoke `result` when it is cancelled, then it can be switched to Color and `result` can just be ignored.

Reply:
Would using [`Color?`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types) work?

Reply:
It should since we updated unity